messages/src/msg_fields/protocols/cred_issuance/mod.rs

```diff
@@ -1,218 +1,26 @@
-                OfferCredential::deserialize(deserializer).map(From::from)
-            }
-            CredentialIssuanceTypeV1_0::ProposeCredential => {
-                ProposeCredential::deserialize(deserializer).map(From::from)
-            }
-            CredentialIssuanceTypeV1_0::RequestCredential => {
-                RequestCredential::deserialize(deserializer).map(From::from)
-            }
-            CredentialIssuanceTypeV1_0::IssueCredential => {
-                IssueCredential::deserialize(deserializer).map(From::from)
-            }
-            CredentialIssuanceTypeV1_0::Ack => {
-                AckCredential::deserialize(deserializer).map(From::from)
-            }
-            CredentialIssuanceTypeV1_0::ProblemReport => {
-                CredIssuanceProblemReport::deserialize(deserializer).map(From::from)
-            }
-            CredentialIssuanceTypeV1_0::CredentialPreview => {
-                Err(utils::not_standalone_msg::<D>(kind_str))
-            }
-        }
-    }
-
-    fn delayed_serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-    where
-        S: Serializer,
-    {
-        match self {
-            Self::OfferCredential(v) => MsgWithType::from(v).serialize(serializer),
-            Self::ProposeCredential(v) => MsgWithType::from(v).serialize(serializer),
-            Self::RequestCredential(v) => MsgWithType::from(v).serialize(serializer),
-            Self::IssueCredential(v) => MsgWithType::from(v).serialize(serializer),
-            Self::Ack(v) => MsgWithType::from(v).serialize(serializer),
-            Self::ProblemReport(v) => MsgWithType::from(v).serialize(serializer),
-        }
-    }
+    V1(CredentialIssuanceV1),
+    V2(CredentialIssuanceV2),
 }
 
-#[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
-pub struct CredentialPreview {
-    #[serde(rename = "@type")]
-    msg_type: CredentialPreviewMsgType,
-    pub attributes: Vec<CredentialAttr>,
-}
-
-impl CredentialPreview {
-    pub fn new(attributes: Vec<CredentialAttr>) -> Self {
-        Self {
-            msg_type: CredentialPreviewMsgType,
-            attributes,
-        }
+impl From<CredentialIssuanceV1> for AriesMessage {
+    fn from(value: CredentialIssuanceV1) -> Self {
+        AriesMessage::CredentialIssuance(CredentialIssuance::V1(value))
     }
 }
 
-/// Non-standalone message type.
-/// This is only encountered as part of an existent message.
-/// It is not a message on it's own.
-#[derive(Copy, Clone, Debug, Default, Deserialize, PartialEq)]
-#[serde(try_from = "CowStr")]
-struct CredentialPreviewMsgType;
-
-impl<'a> From<&'a CredentialPreviewMsgType> for CredentialIssuanceTypeV1_0 {
-    fn from(_value: &'a CredentialPreviewMsgType) -> Self {
-        CredentialIssuanceTypeV1_0::CredentialPreview
+impl From<CredentialIssuanceV2> for AriesMessage {
+    fn from(value: CredentialIssuanceV2) -> Self {
+        AriesMessage::CredentialIssuance(CredentialIssuance::V2(value))
     }
 }
```

Comment:
If you use the `transit_to_aries_msg` variadic arguments this becomes redundant. You should rarely need to work with intermediate structures like this (perhaps only when matching on an `AriesMessage`).

Other than that, the conversion is really just implemented for concrete messages (`MsgParts<RequestCredentialV2Content, RequestCredentialV2Decorators>`) to `AriesMessage`.

Reply:
fixed 👍